pkg/sync/sync_service.go

```diff
@@ -294,6 +295,7 @@ func (syncService *SyncService[H]) setFirstAndStart(ctx context.Context, peerIDs
 //
 // `store` is closed last because it's used by other services.
 func (syncService *SyncService[H]) Stop(ctx context.Context) error {
+	syncService.topicSubscription.Cancel()
```

Comment:
Without unsubscribe, the stop fails on the topic close.

Reply:
should we add that as a code comment?